src/jetson/main.py

```diff
@@ -227,54 +225,62 @@ def classifyFrame(self,
-                img: np.ndarray):
-    '''
-    This function Encrypts faces
-    Args:
-        coordinates - Face coordinates returned by face detector
-        img - A 3D numpy array containing image to be encrypted
-
-    Return:
-        encryptedImg - Image with face coordinates encrypted
-    '''
-
-    encryptor = Encryptor()
-    encryptedImg, _ = encryptor.encrypt(coordinates, img)
-
-    return encryptedImg
-
-def encryptFrame(img:np.ndarray,
-                boxes:List[Tuple[np.float64]]):
-    '''
-    This method takes the face coordinates, encrypts the facial region, writes encrypted image to file filesystem
-    Args:
-        img: A 3D numpy array containing image to be encrypted
-        boxes: facial Coordinates
-    '''
-    try:
+    def encryptFace(self, coordinates: List[Tuple[int]],
+                    img: np.ndarray):
+        '''
+        This function Encrypts faces
+        Args:
+            coordinates - Face coordinates returned by face detector
+            img - A 3D numpy array containing image to be encrypted
+    
+        Return:
+            encryptedImg - Image with face coordinates encrypted
+        '''
+    
+        encryptedImg, _ = self.encryptor.encrypt(coordinates, img)
+    
+        return encryptedImg
+    
+    def encryptFrame(self, img:np.ndarray,
+                    boxes:List[Tuple[np.float64]],
+                    main_process):
+        '''
+        This method takes the face coordinates, encrypts the facial region, writes encrypted image to file filesystem
+        Args:
+            img: A 3D numpy array containing image to be encrypted
+            boxes: facial Coordinates
+        '''
         for box in boxes:
             x1, y1, x2, y2 = [int(b) for b in box]
             # draw boxes within the frame
             x1 = max(0, x1)
             y1 = max(0, y1)
             x2 = min(img.shape[1], x2)
             y2 = min(img.shape[0], y2)
-
-            img = encryptFace([(x1, y1, x2, y2)], img)
-
+    
+            img = self.encryptFace([(x1, y1, x2, y2)], img)
+    
         #TODO ftp img to remote
-        #Lets just write img to filesystem for now
-        global fileCount
-        face_file_name = os.path.join(args.output_dir, f'{fileCount}.jpg')
-
-        #TODO: Remove this print statement after db integration
-        print("writing ", face_file_name)
-        fileCount += 1
-        cv2.imwrite(face_file_name, img)
-    except KeyboardInterrupt:
-        pass
+        if args.write_imgs:
+            if not os.path.isdir(args.output_dir):
+                os.mkdir(args.output_dir)
+            global fileCount
+            face_file_name = os.path.join(args.output_dir, f'{fileCount.value}.jpg')
+    
+            #TODO: Remove this print statement after db integration
+            print("writing ", face_file_name)
+            with fileCount.get_lock():
+                fileCount.value += 1
+            cv2.imwrite(face_file_name, img)
 
 
 if __name__ == "__main__":
```

Comment:
please see [this comment](https://github.com/PurdueCAM2Project/Embedded2/pull/8#pullrequestreview-437686539) for a good example of how to structure scripts/ main files and try and apply that here. its very messy and somewhat hard to follow at what is going on here